src/lib/libast/string/fmtperm.c

```diff
@@ -19,24 +19,25 @@
  *                     Phong Vo <phongvo@gmail.com>                     *
  *                                                                      *
  ***********************************************************************/
-/*
- * Glenn Fowler
- * AT&T Bell Laboratories
- *
- * return strperm() expression for perm
- */
+//
+// Glenn Fowler
+// AT&T Bell Laboratories
+//
+// Return strperm() expression for perm
+//
 #include "config_ast.h"  // IWYU pragma: keep
 
 #include <sys/stat.h>
 
+// This function always returns same static buffer
+// TODO: Fix this function to not return same static buffer
```

Comment:
Same concern I had for `fmtmode()`. Having it return a static buffer is a reasonable choice. The caller can always `strdup()` the result if they need to call this a second time before consuming the result of the first call. Using `fmtbuf()` is dangerous -- we don't want more uses of that API. And having this return a `malloc()`'d buffer just means the caller has to always free it.